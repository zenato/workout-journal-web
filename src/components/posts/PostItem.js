import moment from 'moment';
import React from 'react';

const EventItem = ({ item, onDetail }) => (
  <li key={item.id} className="list-item">
    <a href="" onClick={(e) => onDetail(e, item.id)}>
      <div className="info">
        <span className="name">{moment(item.workout_date).format('YYYY-MM-DD')}</span>
        <span className="weekday">{moment(item.workout_date).format('ddd')}</span>
        <div className="remark">{item.performances.map(p => p.event.name).join(' / ')}</div>
      </div>
    </a>
  </li>
);

export default EventItem;
