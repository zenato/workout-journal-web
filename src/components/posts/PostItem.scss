@import 'common';

.list-item {
  border: solid 1px #c7c7c7;
  margin-bottom: 5px;
  background-color: #fff;

  >a {
    padding: 10px;
    display: flex;

    &:hover {
      background-color: #efefef;
    }
    &:active {
      background-color: #d7d7d7;
    }
  }

  .info {
    flex: 1;
  }

  .name {
    font-weight: bold;
    color: #ff5008;
  }

  .weekday {
    margin-left: 3px;
    font-size: 0.7rem;
    font-weight: bold;
    color: #7c7c7c;
  }

  .value-unit {
    margin-left: 5px;
    color: #4e4e4e;
  }

  .remark {
    color: #a0a0a0;
    font-size: 0.7rem;
    margin-top: 5px;
  }
}
