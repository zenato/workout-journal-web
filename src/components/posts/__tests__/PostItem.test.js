import React from 'react';
import { shallow } from 'enzyme';
import PostItem from '../PostItem';

const item = {
  id: 1,
  workout_date: '2017-08-18T05:54:17.061Z',
  remark: 'test',
  performances: [
    {
      event: {
        name: 'test',
      },
    },
  ],
};

it('renders without error', () => {
  shallow(
    <PostItem
      item={item}
      onDetail={() => {}}
    />
  );
});

it('simulate click link', () => {
  let detailId = -1;
  const wrapper = shallow(
    <PostItem
      item={item}
      onDetail={(e, id) => detailId = id}
    />
  );
  wrapper.find('a').simulate('click');
  expect(detailId).toEqual(1);
});