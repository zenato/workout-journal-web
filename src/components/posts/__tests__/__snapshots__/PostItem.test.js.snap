// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem should render without error. 1`] = `
<styled.li>
  <Link
    href=""
    onClick={[Function]}
  >
    <styled.div>
      <styled.span>
        2017-08-18
      </styled.span>
      <styled.span>
        Fri
      </styled.span>
      <styled.div>
        test
      </styled.div>
    </styled.div>
  </Link>
</styled.li>
`;
